### DISCUSSION TITLE
Terminal runner was still useful?

### DISCUSSION THREAD

**Discussion by dorkster100** [adalessa/laravel.nvim#42] [General]
I may not fully understand, but doesn't the terminal runner have some things the buffer runner cannot do? For example after the completion of the task any button press closes the buffer, so i cannot copy the text from it or scroll.

Also when running tinker command it asks me to include something - which is kind of weird?

I may as well be ignorant, if you are using the buffer runner in some cool ways, let me know, I may not know how to use it properly.

**Comment by adalessa** [comment]
The main problem that I found with the terminal was related on how escape characters. If you need to pass `\\` as part of the namespace was not possible. This generate error on command executions, not being able to handle inputs and commands expect it in the command execution.

For the problem that you mentioned about closing, this is consider in the same way a term buffer, you can exit the terminal mode with the native `Ctrl + \  Ctrl + n` that will put you in normal mode and you can copy and do everything that you want.

Related to the include, yes is a bug,  is related to that tinker has an optional argument `include` and is asking for that since is an argument.  I could not ask for optional arguments, or allow to empty response in optional argument. I am more incline to do the second one. This is when running from telescope.

**Reply by dorkster100** [reply]
I agree about second option and maybe specifying that this is an <optional> argument.

I will try the solution you proposed and will try to run with the current solution, will get back to you eventually, thank you.

**Comment by adalessa** [comment]
For exiting the terminal mode I have a simple mapping `vim.keymap.set("t", "<Esc><Esc>", "<C-\\><C-n>")` could help you as example to something you may found confortable.

**Reply by dorkster100** [reply]
yes, i had this for the terminal, but for this buffer runner it is not working, i will check why later.
Also i found an issue with the buffer runner and multiline input. For example when in artisan db i paste in a big query, when going up the line it gets confused.

**Reply by adalessa** [reply]
I will check that. I didn't test it.

**Reply by adalessa** [reply]
I found the problem, and develop a fix here #44 
The problem is how the data was being sent. When the up key is sent to the process the back process returned with a single string containing the multiline information in just one with the break lines these were not property handle by the native functions. So I had to build a parser that check for the break lines and send to the buffer to proper visualization. 
This does not modify the information being sent just acts as a middleware  for the buffer. 
Please test and let me know.

**Reply by dorkster100** [reply]
I don't think it fixed my issue, i will elaborate.
Steps to reproduce:
Open artisan db
Copy a large query and paste it into the db buffer
Now suppose you want to add `explain` to the beginning - once you go into Terminal mode, you will be teleported to the end of the line, now to get back, you start hitting left arrow, once it reaches beggining of the current line in the buffer - it stops and if you proceed clicking left arrow after that everything becomes a mess. The cursor is not in the correct place anymore, how it is displayed and where it actually is - is different

**Reply by adalessa** [reply]
I have seen the issue, and yes for this case the buffer runner is not the proper solution, for simple one liners is ok but not for big queries. 
I have reverted the removal and add it #46. 
I added in the dock that is not possible to escape characters in the command, that was a problem that I was having.

**Comment by adalessa** [comment]
#43 fixes the optional parameter
